index.js

```diff
@@ -80,23 +82,46 @@ fetch(apiUrl)
         iconArray.push(data.list[2].weather[0].icon);
         iconArray.push(data.list[10].weather[0].icon);
         iconArray.push(data.list[18].weather[0].icon);
-    })
 
-/* tried to fetch api url from flicker for changing background images,but encountering with an error */
+        function createIcon() {
+            let iconSet = document.querySelectorAll("ul > li");
+            for (let i = 0; i < iconArray.length && i < iconSet.length; i++) {
+                let newIconEle = document.createElement("span");
+                newIconEle.classList.add("iconStyle");
+                let imageUrl = "http://openweathermap.org/img/wn/" + iconArray[i] + "@2x.png";
+                newIconEle.innerHTML = "<img src=" + imageUrl + ">";
+                iconSet[i].appendChild(newIconEle);
+            }
+        }
+        createIcon();
+    });
 searchBtnCmp.addEventListener("click", function() {
-
-    const cityUrl = "http://api.flickr.com/services/rest/?method=flickr.photos.search&api_key=528fc5e2ec09109cd0696b0555e7bc81&tags=" + weatherSearchBar.value + "&format=json&nojsoncallback=1";
+    const proxyurl = "https://cors-anywhere.herokuapp.com/";
```

Comment:
you don't need this magic proxy.
what you need to do is to use `https` protocol for your api instead of `http`
`const cityUrl = "https://api.flickr.com/services/rest/?method=flickr.photos.search&api_key=ef3d8a6207b554e853fcfe4b900bd9b7&tags=" + weatherSearchBar.value + "&format=json&nojsoncallback=1";`
and this would solve CORS problem.